smap_l2_gridder/crs.py

```diff
@@ -0,0 +1,259 @@
+}
+
+
+def geotransform_from_target_info(target_info: dict) -> Geotransform:
+    """Return a geotransform from the grid_info dict.
+
+    grid_info contains a parsed NSIDC gpd file.
+
+    GT(0) x-coordinate of the upper-left corner of the upper-left pixel.
+    GT(1) w-e pixel resolution / pixel width.
+    GT(2) row rotation (typically zero).
+    GT(3) y-coordinate of the upper-left corner of the upper-left pixel.
+    GT(4) column rotation (typically zero).
+    GT(5) n-s pixel resolution / pixel height (negative value for a north-up image).
+
+    """
+    validate_gpd_style(target_info)
+    return Geotransform(
+        target_info['Map Origin X'],
+        target_info['Grid Map Units per Cell'],
+        np.float64(0.0),
+        target_info['Map Origin Y'],
+        np.float64(0.0),
+        -1.0 * target_info['Grid Map Units per Cell'],
+    )
+
+
+def validate_gpd_style(target_info: dict) -> None:
+    """Raise error if the gpd is non-standard."""
+    if (target_info['Grid Map Origin Column'] != -0.5) or (
+        target_info['Grid Map Origin Row'] != -0.5
+    ):
+        raise InvalidGPDError('Can not use non standard gpd.')
+
+
+def convert_value(value: str) -> str | np.float64 | np.long:
+    """Convert gpd value to appropriate type."""
+    value = value.strip()
+    try:
+        if '.' in value:
+            return np.float64(value)
+        else:
+            return np.long(value)
+    except ValueError:
+        return value
+
+
+def parse_gpd_file(gpd_name: str) -> dict:
+    """Parse a grid parameter definition file and return a dictionary of parameters.
+
+    Parameters:
+    -----------
+    filename : str
+        Full path to a grid parameter definition file
+       or
+        Name of a gpd file in the references directory.
+
+    Returns:
+    --------
+    dict
+        Dictionary containing parameter names as keys and their values as values.
+        Decimal numbers are converted to np.float64
+        Integer-like numbers are converted to np.long
+        Non-numeric values remain as strings
+
+    """
+    gpd_info = {}
+
+    if Path(gpd_name).exists():
+        filename = gpd_name
+    else:
+        filename = f'{Path(__file__).parent}/reference/{gpd_name}'
+
+    with open(filename, encoding='utf-8') as f:
+        for line in f:
+
+            line = line.strip()
+            if not line or line.startswith(';'):
+                continue
+
+            if ':' in line:
+                key, value = line.split(':', 1)
+                key = key.strip()
+                # Remove comments and strip whitespace
+                value = value.split(';')[0].strip()
+
+                if key and value:
+                    gpd_info[key] = convert_value(value)
+
+    return gpd_info
+
+
+def create_crs(target_info: dict) -> DataArray:
+    """Return the correct CRS for the target grid."""
+    crs = CRS(target_info['wkt'])
+    the_crs = xr.DataArray(data=b'', attrs={**crs.to_cf(), 'proj': crs.to_proj4()})
```

Comment:
Just means it's an empty byte object.  There's no data associated with the CRS, it's just there to hang metadata on.  It could have been just `data=''` but it's just slightly smaller.  I was trying to use a `char` but can't in python.